File: PARADIGMAS/1608/main.py
def get_lowest_price(b, ingredients_prices):
    lowest_price = 100 * 1000 * 1000
    for _ in range(b):
        cake_price = _get_cake_price(ingredients_prices)
        if cake_price < lowest_price:
            lowest_price = cake_price
    return lowest_price


def _get_cake_price(ingredients_prices):
    price = 0
    qi, *recipe = list(map(int, input().split()))
    for i in range(qi):
        ingredient = recipe[i * 2]
        quantity = recipe[i * 2 + 1]
        price += ingredients_prices[ingredient] * quantity
    return price


t = int(input())
for _ in range(t):
    d, _, b = tuple(map(int, input().split()))
    ingredients_prices = list(map(int, input().split()))
    lowest_price = get_lowest_price(b, ingredients_prices)
    print(d // lowest_price)

File: PARADIGMAS/1608/main.js
function getLowestPrice(b, ingredientsPrices) {
    let lowestPrice = 100 * 1000 * 1000;
    for (let cake = 0; cake < b; cake++) {
        let cakePrice = _getCakePrice(ingredientsPrices);
        if (cakePrice < lowestPrice) {
            lowestPrice = cakePrice;
        }
    }
    return lowestPrice;
}

function _getCakePrice(ingredientsPrices) {
    let price = 0;
    let [qi, ...recipe] = lines[i++].split(' ').map(Number);
    for (let i = 0; i < qi; i++) {
        let ingredient = recipe[i * 2];
        let quantity = recipe[i * 2 + 1];
        price += ingredientsPrices[ingredient] * quantity;
    }
    return price;
}

let input = require('fs').readFileSync('/dev/stdin', 'utf-8');
var lines = input.split('\n');
var i = 0;
let t = parseInt(lines[i++]);
for (let times = 0; times < t; times++) {
    let [d, _, b] = lines[i++].split(' ').map(Number);
    let ingredientsPrices = lines[i++].split(' ').map(Number);
    let lowestPrice = getLowestPrice(b, ingredientsPrices);
    console.log(parseInt(d / lowestPrice));
}